### PR TITLE
Async client creation

### PR DESCRIPTION
**Pull request by jsnoble**
- base foundation to create async elasticsearch client (only enough for spaces)
- updated packages

### CODE REVIEW

File: packages/elasticsearch-store/src/elasticsearch-client/create-client.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        if (body.version.distribution === 'opensearch') {
+            // TODO: clean this up
+            const openConfig = {
+                ...config,
+                node: gotOptions.url as string
+            };
+            const client = new open.Client(openConfig);
+            const meta = {
+                client_type: 'opensearch',
```
Comment (godber) [inline packages/elasticsearch-store/src/elasticsearch-client/create-client.ts:54]: Change `client_type` to `distribution`.

**Review by godber — CHANGES_REQUESTED** [review]
We reviewed this on zoom and have requested the changes listed here as well as changing the conditionals on version to version and `distribution`.

File: packages/teraslice/lib/storage/backends/elasticsearch_store.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ module.exports = function elasticsearchStorage(backendConfig) {
 
         if (fields) {
             const esVersion = elasticsearch.getESVersion();
-            if (esVersion > 6) {
+            if (esVersion !== 6) {
```
Comment (kstaken) [inline packages/teraslice/lib/storage/backends/elasticsearch_store.js:87]: Does this need to include a check on the `distribution` as well?

Comment (jsnoble) [reply packages/teraslice/lib/storage/backends/elasticsearch_store.js:87]: hmm, good spot you found, is should be able to use the helper method isElasticsearch6 for this

**Pull request merged**